### PR TITLE
I1697

### PR DESCRIPTION
**Pull request by littledice**
This ended up being more work than I originally anticipated, because I wanted a way to run the app locally with a different logging configuration. This is handled by having two gradle tasks, `copyDockerProperties` and `copyDevProperties` that are run conditionally depending on whether the `prod` system property is set (which it is in the Dockerfile). 

A side benefit is that this `prod` property in then also accessible by the kotlin code, so it can look for a config file in a test location rather than at an absolute path when running  locally; see the `EnvironmentProperties` object for this logic.

1. Write logs to volume when running in container
1. Write logs to stdout when running locally
1. gradle task for running locally `runAppLocally`

### CODE REVIEW

File: src/test/kotlin/org/vaccineimpact/kodiak/tests/KodiakTests.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import com.nhaarman.mockito_kotlin.never
+import com.nhaarman.mockito_kotlin.verify
+import org.junit.Before
+import org.junit.Test
+import org.slf4j.Logger
+import org.vaccineimpact.kodiak.JsonConfig
+import org.vaccineimpact.kodiak.Kodiak
+
+class KodiakTests : BaseTests() {
+
```
Comment (littledice) [inline src/test/kotlin/org/vaccineimpact/kodiak/tests/KodiakTests.kt]: Obviously the functionality here is really yet to be written, but this is really to demonstrate the testing pattern we can use

File: src/main/resources/dev/log4j.properties [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# For the general syntax of property based configuration files see
```
Comment (littledice) [inline src/main/resources/dev/log4j.properties]: only log to stdout when running locally

File: src/main/resources/dev/config.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (littledice) [inline src/main/resources/dev/config.json]: test config for running locally

File: src/main/kotlin/org/vaccineimpact/kodiak/AppConfig.kt [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package org.vaccineimpact.kodiak
+
+import java.io.File
+import java.io.InputStream
+
+object AppConfig {
+
+    private val prod = System.getProperty("prod", "false").toBoolean()
```
Comment (littledice) [inline src/main/kotlin/org/vaccineimpact/kodiak/AppConfig.kt]: The idea is to set this property to `true` when building the dockerised app. See https://github.com/vimc/kodiak/pull/5/files#diff-3254677a7917c6c01f55212f86c57fbfR44

File: build.gradle [hunk 5]
```diff
@@ -65,4 +66,43 @@ distDocker {
...
+    outputs.upToDateWhen { false }
+}
+
+task copyDevProperties(type: Copy) {
+    from 'src/main/resources/dev'
+    into 'src/main/resources'
+    outputs.upToDateWhen { false }
+}
+
+if (System.getProperty("prod")) {
```
Comment (MartinEden) [inline build.gradle]: Would recommend you use "production" rather than "prod", for readability.

File: build.gradle [hunk 6]
```diff
@@ -65,4 +66,43 @@ distDocker {
...
+
+// this method creates a string that can be used by Eval
+static def setCommandLineArguments(commandLineArgs) {
+
+    def arguments = commandLineArgs.tokenize()
+    def cla = "["
+    arguments.each {
+        cla += "'" + "${it}" + "',"
+    }
+    return cla.substring(0, cla.lastIndexOf(',')) + "]"
```
Comment (MartinEden) [inline build.gradle]: Can you not return a groovy list here, and then use something like `args setCommandLineArguments(commandLineArgs)`? Is the Eval necessary?

File: src/main/kotlin/org/vaccineimpact/kodiak/EnvironmentProperties.kt [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+object EnvironmentProperties {
+
+    val prod = System.getProperty("prod", "false").toBoolean()
+
+    val configSource: InputStream = if (prod) {
+        File("/etc/kodiak/config.json").inputStream()
+    } else {
+        EnvironmentProperties::class.java.classLoader.getResource("config.json")
+                .openStream()
+    }
```
Comment (MartinEden) [inline src/main/kotlin/org/vaccineimpact/kodiak/EnvironmentProperties.kt]: Do these streams get closed/disposed anywhere?

File: build.gradle (hunk 6 above)
Comment (littledice) [reply build.gradle]: Huh, good point. Totally unnecessary!

**Comment by littledice** [comment]
All changes made. Also realised that the system property "prod" was not actually making it through from the build container to the final kodiak image. Instead have used the `setEnvironment` option available in the `distDocker` task.

**Pull request merged**